src/core/tokenize/match.ts

```diff
@@ -0,0 +1,206 @@
+import { Token } from '../classes/token';
+import singleton from '../prism';
+import { tokenize } from './tokenize';
+import { resolve } from './util';
+import type { GrammarToken, GrammarTokens, RegExpLike } from '../../types';
+import type {
+	LinkedList,
+	LinkedListHeadNode,
+	LinkedListMiddleNode,
+	LinkedListTailNode,
+} from '../linked-list';
+import type { Prism } from '../prism';
+
+export function _matchGrammar (
+	this: Prism,
+	text: string,
+	tokenList: LinkedList<string | Token>,
+	grammar: GrammarTokens,
+	startNode: LinkedListHeadNode<string | Token> | LinkedListMiddleNode<string | Token>,
+	startPos: number,
+	rematch?: RematchOptions
+): void {
+	const prism = this ?? singleton;
+
+	for (const token in grammar) {
+		const tokenValue = grammar[token];
+		if (!grammar.hasOwnProperty(token) || token === '$tokenize' || !tokenValue) {
```

Comment:
This is where we should handle special properties explicitly: when `tokenize` was a symbol, we didn't have to do that since it wasn't enumerable. Now, it shouldn't be matched as regular tokens. Probably because we don't do that in the `simplify` branch, we have tests that fail, and before, I couldn't understand why. Now I know. 🙂

Reply:
Done! Thanks.